ManagementPacks/Community.DataOnDemand/Scripts/Get-Processes.ps1

```diff
@@ -52,25 +53,36 @@ foreach ($PoshProcess in $PoshProcesses)
     }
 
     # Create a set of output objects with properties from WMI and posh
-    $OutputObjects += New-Object -TypeName PSObject -Property @{
-        "Pid"=$PoshProcess.Id;
-        "ParentPid"=$WmiProcess.CreatingProcessID;
-        "SessionId"=$PoshProcess.SessionId;
-        "Name"=$PoshProcess.Name;
-        "Description"=$PoshProcess.Description;
-        "Path"=$PoshProcess.Path;
-        "CpuPercent"=$WmiProcess.PercentProcessorTime;
-        "PrivateBytes"=$WmiProcess.PrivateBytes;
-        "Handles"=$WmiProcess.HandleCount;
-        "Threads"=$WmiProcess.ThreadCount;
-    }
+    $OutputObject = New-Object -TypeName PSObject
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Pid -Value $PoshProcess.Id
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Name -Value $PoshProcess.Name
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name CpuPercent -Value $WmiProcess.PercentProcessorTime
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name PrivateBytes -Value $WmiProcess.PrivateBytes
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Description -Value $PoshProcess.Description
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name ParentPid -Value $WmiProcess.CreatingProcessID
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name SessionId -Value $PoshProcess.SessionId
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Handles -Value $WmiProcess.HandleCount
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Threads -Value $WmiProcess.ThreadCount
+    Add-Member -InputObject $OutputObject -MemberType NoteProperty -Name Path -Value $PoshProcess.Path
+
+    $OutputObjects += $OutputObject
+    # Do not null OutputObject object it is used further down to extract correct ordering of object properties
```

Comment:
Good question, a lack of imagination at the time. In hindsight this is a better solution and one I should have thought of.